src/TiltObserver.cpp

```diff
@@ -0,0 +1,227 @@
+  imuVelC_.angular() = anchorImuKine.angVel();
+
+  // update the pose as PTransformd for the logs
+  X_C_IMU_.translation() = anchorImuKine.position();
+  X_C_IMU_.rotation() = anchorImuKine.orientation.toMatrix3().transpose();
+
+  /* Update of the observed robot */
+  my_robots_->robot().mbc().q = ctl.realRobot().mbc().q;
+  update(my_robots_->robot(), ctl);
+
+  return true;
+}
+
+void TiltObserver::update(mc_control::MCController & ctl)
+{
+  if(updateRobot_)
+  {
+    auto & robot = ctl.realRobots().robot(updateRobot_);
+    auto posW = robot.posW();
+    R_0_fb_ = estimatedRotationIMU_ * fbImuKine_.orientation.toMatrix3();
+    posW.rotation() = R_0_fb_.transpose();
+    mc_rtc::log::info("update robot: {}", mc_rbdyn::rpyFromMat(posW.rotation()));
+    posW.translation() = Eigen::Vector3d::Zero();
+    robot.posW(posW);
+    robot.forwardKinematics();
+  }
+
+  if(updateSensor_)
+  {
+    // auto & imu = ctl.robot(robot_).bodySensor(imuSensor_);
+    auto & imu = const_cast<mc_rbdyn::BodySensor &>(ctl.robot(robot_).bodySensor(imuSensor_));
+    auto & rimu = const_cast<mc_rbdyn::BodySensor &>(ctl.realRobot(robot_).bodySensor(imuSensor_));
+
+    imu.orientation(Eigen::Quaterniond{estimatedRotationIMU_});
+    rimu.orientation(Eigen::Quaterniond{estimatedRotationIMU_});
+  }
+}
+
+void TiltObserver::update(mc_rbdyn::Robot & robot, const mc_control::MCController & ctl)
+{
+  const auto & realRobot = ctl.realRobot(robot_);
+  R_0_fb_ = estimatedRotationIMU_ * fbImuKine_.orientation.toMatrix3();
+  poseForDisplay = realRobot.posW();
+  poseForDisplay.rotation() = R_0_fb_.transpose();
+  robot.posW(poseForDisplay);
+}
+
+void TiltObserver::addToLogger(const mc_control::MCController &, mc_rtc::Logger & logger, const std::string & category)
+{
+  logger.addLogEntry(category + "_imuVelC", [this]() -> const sva::MotionVecd & { return imuVelC_; });
+  logger.addLogEntry(category + "_imuPoseC", [this]() -> const sva::PTransformd & { return X_C_IMU_; });
+  logger.addLogEntry(category + "_imuEstRotW", [this]() { return Eigen::Quaterniond{estimatedRotationIMU_}; });
+  logger.addLogEntry(category + "_controlAnchorFrame", [this]() -> const sva::PTransformd & { return X_0_C_; });
+  logger.addLogEntry(category + "_displayedWorldFbPose",
+                     [this]() -> const sva::PTransformd & { return poseForDisplay; });
+  logger.addLogEntry(category + "_anchorFramelinVel_global",
+                     [this]() -> const so::Vector3 & { return worldAnchorLinVel; });
+  logger.addLogEntry(category + "_anchorFramelinVel_local",
+                     [this]() -> const so::Vector3 & { return worldAnchorLocalLinVel; });
+}
+
+void TiltObserver::removeFromLogger(mc_rtc::Logger & logger, const std::string & category)
+{
+  logger.removeLogEntry(category + "_imuVelC");
+  logger.removeLogEntry(category + "_imuPoseC");
+  logger.removeLogEntry(category + "_imuEstRotW");
+  logger.removeLogEntry(category + "_controlAnchorFrame");
```

Comment:
This is missing a few entries. The simplest approach here is to add the observer as the source for the log entries, so for example:
```cpp
logger.addLogEntry(category + "_imuVelC", [this]() -> const sva::MotionVecd & { return imuVelC_; });
```
becomes
```cpp
logger.addLogEntry(category + "_imuVelC", this, [this]() -> const sva::MotionVecd & { return imuVelC_; });
```
then `removeFromLogger` can be implemented as:
```cpp
logger.removeLogEntries(this);
```